### PR TITLE
Correct German scraper

### PR DESCRIPTION
**Pull request by mathiasbynens**

### CODE REVIEW

File: src/covid_updater/scraping/countries/germany.py [hunk 1]
```diff
@@ -15,16 +15,16 @@ def __init__(self):
             },
             column_renaming={
                 "state": "region",
-                "initialDosesCumulative": "people_vaccinated",
-                "finalDosesCumulative": "people_fully_vaccinated",
+                "atLeastPartiallyVaccinatedCumulative": "people_vaccinated",
```
Comment (mathiasbynens) [inline src/covid_updater/scraping/countries/germany.py:18]: I'm not sure which metric you want exactly: is it `atLeastPartiallyVaccinatedCumulative` or `onlyPartiallyVaccinatedCumulative`? See https://github.com/mathiasbynens/covid-19-vaccinations-germany#partially-vs-fully-vaccinated. Let me know and I'd be happy to update the patch.

Comment (lucasrodes) [reply src/covid_updater/scraping/countries/germany.py:18]: Hi, and - once again - thanks @mathiasbynens!

I am currently reporting three metrics (same as OWID):
- `total_vaccinations` := Cumulative number of doses administered.
- `people_vaccinated` := Cumulative number of people that received >= 1 dose (either first one in 2-dose-scheme doses or one and only in 1-dose-scheme).
- `people_fully_vaccinated` := Cumulative number of people that are fully vaccinated (received 2 doses in 2-dose-scheme or 1 dose in 1-dose scheme).

Having a look at your definitions, I'd say that:

```
total_vaccinations = onlyPartiallyVaccinatedCumulative + fullyVaccinatedCumulative
people_vaccinated = atLeastPartiallyVaccinatedCumulative
people_fully_vaccianted = fullyVaccinatedCumulative
```

Therefore some small modifications in the patch would be required. I'll point them as comments

File: src/covid_updater/scraping/countries/germany.py [hunk 2]
```diff
@@ -15,16 +15,16 @@ def __init__(self):
...
+                "fullyVaccinatedCumulative": "people_fully_vaccinated",
             },
         )
 
     def load_data(self):
         # Load
         df = pd.read_csv(
             self.data_url,
-            usecols=["date", "state", "initialDosesCumulative", "finalDosesCumulative"],
+            usecols=["date", "state", "atLeastPartiallyVaccinatedCumulative", "fullyVaccinatedCumulative"],
```
Comment (lucasrodes) [inline src/covid_updater/scraping/countries/germany.py:27]: Include `onlyPartiallyVaccinatedCumulative`, later required to estimate later `total_vaccinations`

**Review by lucasrodes — CHANGES_REQUESTED** [review]
Additionally, changes to function `_process` are needed.

I can deal with these if you prefer. But basically:

```python
df.loc[:, "total_vaccinations"] = (
    df.loc[:, "onlyPartiallyVaccinatedCumulative"] + df.loc[:, "people_fully_vaccinated"]
)
df = df.drop(columns=["onlyPartiallyVaccinatedCumulative"])
```

**Comment by mathiasbynens** [comment]
> I can deal with these if you prefer

Oh yeah, in that case please feel free to take over the patch! I think that'd be the quickest way forward. I just wanted to get the ball rolling with this PR :) Thanks!

**Comment by lucasrodes** [comment]
@mathiasbynens Just as a sanity check and to align on metrics:

I am currently tracking the following metrics:

- total_vaccinations := Cumulative number of doses administered. Likely equivalent to `initialDosesCumulative` + `finalDosesCumulative`
- people_vaccinated := Cumulative number of people that received >= 1 dose (either first one in 2-dose-scheme doses or one and only in 1-dose-scheme). Likely equivalent to `atLeastPartiallyVaccinatedCumulative`
- people_fully_vaccinated := Cumulative number of people that are fully vaccinated (received 2 doses in 2-dose-scheme or 1 dose in 1-dose scheme). Likely equivalent to `finalDosesCumulative`

**Comment by mathiasbynens** [comment]
> @mathiasbynens Just as a sanity check and to align on metrics:
> 
> I am currently tracking the following metrics:
> 
> * `total_vaccinations` := Cumulative number of doses administered. Likely equivalent to `initialDosesCumulative + finalDosesCumulative` or `initialDosesCumulative + fullyVaccinatedCumulative`
> * `people_vaccinated` := Cumulative number of people that received >= 1 dose (either first one in 2-dose-scheme doses or one and only in 1-dose-scheme). Likely equivalent to `atLeastPartiallyVaccinatedCumulative`
> * `people_fully_vaccinated` := Cumulative number of people that are fully vaccinated (received 2 doses in 2-dose-scheme or 1 dose in 1-dose scheme). Likely equivalent to `finalDosesCumulative` (or `fullyVaccinatedCumulative`)

Can confirm! This sounds correct.

**Comment by lucasrodes** [comment]
Thanks for your help @mathiasbynens

**Comment by mathiasbynens** [comment]
Resulting diff: https://github.com/sociepy/covid19-vaccination-subnational/commit/cd406104a0fda812fb8768e51e1dbeea0b3add9b#diff-457e6c6c6497635e86314140f2885d4d3f9347f1b734529613bf2844cecbb362 (J&J shots were previously not included in `people_vaccinated`.)

**Pull request merged**